spec/integration/runner/variant-reassignment-example-13.spec.js

```diff
@@ -0,0 +1,88 @@
+ |         |                            | variants: [                           |                    |                            |                                                             |
+ |         |                            | { id: 2, sku: v2 },{ id: 3, sku: v3 } |                    |                            |                                                             |
+ |         |                            | ]                                     |                    |                            |                                                             |
+ +---------+----------------------------+---------------------------------------+--------------------+----------------------------+-------------------------------------------------------------+
+ */
+/* eslint-enable max-len */
+describe('Variant reassignment', () => {
+  const logger = utils.createLogger(__filename)
+  let ctpClient
+  let product1
+
+  before(async () => {
+    ctpClient = await utils.createClient()
+
+    const results = await utils.createCtpProducts([['1', '2', '3']], ctpClient)
+    product1 = results[0]
+    let response = await ctpClient.products
+      .byId(product1.id)
+      .update({
+        version: product1.version,
+        actions: [{ action: 'publish' }]
+      })
+    response = await ctpClient.products
+      .byId(product1.id)
+      .update({
+        version: response.body.version,
+        actions: [
+          { action: 'removeVariant', id: product1.variants[0].id },
+          { action: 'removeVariant', id: product1.variants[1].id }
+        ]
+      })
+    await ctpClient.products.byId(product1.id).update({
+      version: response.body.version,
+      actions: [
+        { action: 'addVariant', sku: product1.variants[0].sku },
+        { action: 'addVariant', sku: '4' }
+      ]
+    })
+  })
+
+  after(async () => {
+    await utils.unpublishAllProducts(ctpClient, logger)
+    await utils.deleteResourcesAll(ctpClient, logger)
+  })
+
+  it('different product versions on staged and current', async () => {
+    const reassignment = new VariantReassignment([], logger, {})
+    await reassignment.execute([{
+      productType: product1.productType,
+      name: product1.name,
+      slug: product1.slug,
+      masterVariant: product1.masterVariant
+    }], [product1])
+    const { body: product } = await ctpClient.products.byId(product1.id).fetch()
+    expect(product.masterData.current.masterVariant.sku).to.equal('1')
+    expect(product.masterData.current.variants.length).to.equal(0)
+    expect(product.masterData.staged.masterVariant.sku).to.equal('1')
+    expect(product.masterData.staged.variants.length).to.equal(0)
+    const { body: { results } } = await utils.getProductsBySkus(['2', '3', '4'], ctpClient)
+    expect(results).to.have.lengthOf(1)
+    const updatedProductProjection = results[0]
+    expect(updatedProductProjection.masterVariant.sku).to.equal('2')
```

Comment:
should we also check if the product is anonymized with `_ctsd` slug?

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/19/commits/5b95f99660119eacee33f45fed265f97ebcd9aa9